usvm-ts/src/main/kotlin/org/usvm/TSExpressions.kt

```diff
@@ -34,6 +36,95 @@ class TSUndefinedValue(ctx: TSContext) : UExpr<TSUndefinedSort>(ctx) {
     }
 }
 
+/**
+ * Utility class for merging expressions with [UBoolSort] sort.
+ *
+ * Mainly created for [not] function used in TSStateForker.
+ */
+class UJoinedBoolExpr(
+    ctx: TSContext,
+    val exprs: List<UBoolExpr>
+) : UBoolExpr(ctx) {
+    override val sort: UBoolSort
+        get() = ctx.boolSort
+
+    private val joinedExprs = ctx.mkAnd(exprs)
+
+    fun not(): UBoolExpr = ctx.mkAnd(exprs.map(ctx::mkNot))
+
+    override fun accept(transformer: KTransformerBase): KExpr<UBoolSort> {
+        return transformer.apply(joinedExprs)
+    }
+
+    override fun internEquals(other: Any): Boolean = structurallyEqual(other)
+
+    override fun internHashCode(): Int = hash()
+
+    override fun print(printer: ExpressionPrinter) {
+        printer.append("joined(")
+        joinedExprs.print(printer)
+        printer.append(")")
+    }
+}
+
+/**
+ * [UExpr] wrapper that handles type coercion.
+ *
+ * @param value wrapped expression.
+ */
+class TSWrappedValue(
+    ctx: TSContext,
+    val value: UExpr<out USort>,
+    private val scope: TSStepScope
+) : USymbol<USort>(ctx) {
+    override val sort: USort
+        get() = value.sort
+
+    private val transformer = TSExprTransformer(value, scope)
+
+    fun asSort(sort: USort): UExpr<out USort>? = transformer.transform(sort)
+
+    private fun coerce(
+        other: UExpr<out USort>,
+        action: CoerceAction
+    ): UExpr<out USort> = when (other) {
+        is UIntepretedValue -> {
+            val otherTransformer = TSExprTransformer(other, scope)
+            transformer.intersectWithTypeCoercion(otherTransformer, action)
+        }
+
+        is TSWrappedValue -> {
+            transformer.intersectWithTypeCoercion(other.transformer, action)
+        }
+
+        else -> error("Unexpected $other in type coercion")
+    }
+
+    fun coerceWithSort(
+        other: UExpr<out USort>,
+        action: CoerceAction,
+        sort: USort,
+    ): UExpr<out USort> {
+        transformer.transform(sort)
+        return coerce(other, action)
+    }
+
+    override fun accept(transformer: KTransformerBase): KExpr<USort> {
+        return value.cast()
+    }
+
+    override fun internEquals(other: Any): Boolean = structurallyEqual(other)
+
+    override fun internHashCode(): Int = hash()
```

Comment:
Why these methods are not linked with a value?

Reply:
I just ripped off these implementations from some core usvm symbols, will mark it as draft with a comment